metashare/api/jobs.py

```diff
@@ -20,13 +20,13 @@
 logger = logging.getLogger(__name__)
 
 
-def _create_branches_on_github(*, user, repo_url, project, task, repo_root):
+def _create_branches_on_github(*, user, repo_url, repo_id, project, task, repo_root):
     """
     Expects to be called in the context of a local github checkout.
     """
     gh = gh_given_user(user)
-    owner, repo = extract_owner_and_repo(repo_url)
-    repository = gh.repository(owner, repo)
+    owner, name = extract_owner_and_repo(repo_url)
+    repository = gh.repository_with_id(repo_id)
```

Comment:
I don't think there's any reason for `extract_owner_and_repo` to exist anymore. We should consistently use `get_repo_info` (instead of `gh_given_user`) and then refer to `repo.name` and `repo.owner` (instead of `extract_owner_and_repo`).